csharp/GildedRoseTest.cs

```diff
@@ -7,12 +7,185 @@ namespace csharp
+            app.UpdateQuality();
+            
+            Assert.AreEqual(4, Items[0].Quality);
+            Assert.AreEqual(50, Items[1].Quality);
+            Assert.AreEqual(2, Items[2].Quality);
+        }
+        
+        [Test]
+        public void BackstagePassesUnderFiveDays()
+        {
+            IList<Item> Items = new List<Item>();
+            Items.Add(new Item { Name = "Backstage passes thingy", SellIn = 1, Quality = 2 });
+            Items.Add(new Item { Name = "Backstage passes thingy", SellIn = 5, Quality = 5 });
+            
+            GildedRose app = new GildedRose(Items);
+            app.UpdateQuality();
+            
+            Assert.AreEqual(5, Items[0].Quality);
+            Assert.AreEqual(8, Items[1].Quality);
+        }
+        
+        [Test]
+        public void BackstagePassesUnderBetweenFiveAndTenDays()
+        {
+            IList<Item> Items = new List<Item>();
+            Items.Add(new Item { Name = "Backstage passes thingy", SellIn = 6, Quality = 2 });
+            Items.Add(new Item { Name = "Backstage passes thingy", SellIn = 9, Quality = 5 });
+            
+            GildedRose app = new GildedRose(Items);
+            app.UpdateQuality();
+            
+            Assert.AreEqual(4, Items[0].Quality);
+            Assert.AreEqual(7, Items[1].Quality);
+        }
+        
+        [Test]
+        public void BackstagePassesAboveTenDays()
+        {
+            IList<Item> Items = new List<Item>();
+            Items.Add(new Item { Name = "Backstage passes thingy", SellIn = 11, Quality = 2 });
+            Items.Add(new Item { Name = "Backstage passes thingy", SellIn = 16, Quality = 5 });
+            
+            GildedRose app = new GildedRose(Items);
+            app.UpdateQuality();
+            
+            Assert.AreEqual(3, Items[0].Quality);
+            Assert.AreEqual(6, Items[1].Quality);
+        }
+        
+        [Test]
+        public void BackstagePassesSellInNegative()
+        {
+            IList<Item> Items = new List<Item>();
+            Items.Add(new Item { Name = "Backstage passes thingy", SellIn = -1, Quality = 2 });
+            Items.Add(new Item { Name = "Backstage passes thingy", SellIn = -12, Quality = 5 });
+            
+            GildedRose app = new GildedRose(Items);
+            app.UpdateQuality();
+            
+            Assert.AreEqual(0, Items[0].Quality);
+            Assert.AreEqual(0, Items[1].Quality);
+        }
+        
+        [Test]
+        public void Sulfuras()
+        {
+            IList<Item> Items = new List<Item>();
+            Items.Add(new Item { Name = "Sulfuras, Hand of Ragnaros", SellIn = 13, Quality = 2 });
+
+            GildedRose app = new GildedRose(Items);
+            app.UpdateQuality();
+            
+            Assert.AreEqual(12, Items[0].SellIn);
+            Assert.AreEqual(80, Items[0].Quality);
+        }
+        
+        [Test]
+        public void ConjuredPositive()
```

Comment:
This is also a bit vague, don't be afraid to name tests overly specific things, like `ConjuredItemsDecreaseTwiceAsFastInQuality`

Reply:
Although methods shouldn't be as specific, for tests it's easier to figure out what went wrong if a test fails